air/src/proof/context.rs

```diff
@@ -148,7 +165,15 @@ impl Deserializable for Context {
         // read options
         let options = ProofOptions::read_from(source)?;
 
-        Ok(Context { trace_info, field_modulus_bytes, options })
+        // read total number of constraints
+        let total_num_of_constraints = source.read_usize()?;
```

Comment:
Would this not make deserialization tricky in the context of a recursive verifier? Or we don't need to deserialize the context there?

Reply:
Why would that be the case?

Reply:
I see what you mean, I think. 
Currently `Context::to_elements` discards the number of constraints and I think this is fine. We could add also add it at the cost one extra field element that has to be absorbed when initializing the Fiat-Shamir transcript.

Reply:
Adding 1 more element would make `Context::to_elements()` return 9 elements, right? Maybe we can change how `ProofOptions::to_elements()` works so that it takes 1 or 2 elements less, and then adding 1 more element here should be fine.

For `ProofOptions`, we could probably put blowup factor, grinding factor, and the number of queries into a single element.

Reply:
Indeed, it will increase it to 9.
I went ahead with the solution of adding the blowup factor to the field element already encoding field extension, fri folding factor and remainder poly max degree.
The reason for this choice is that it would be nice to have number of queries and grinding factor as field elements so that it is easy to change them, for example, in testing.